projects/cn0561/common/cn0561_bd.tcl

```diff
@@ -0,0 +1,132 @@
+  create_bd_pin -dir I odr
+  create_bd_pin -dir O irq
+  create_bd_intf_pin -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 m_spi
+  create_bd_intf_pin -mode Master -vlnv xilinx.com:interface:axis_rtl:1.0 M_AXIS_SAMPLE
+
+  if {$adc_resolution == 16 || $adc_resolution == 24} {
+    set data_width 32
+  } elseif {$adc_resolution == 32} {
+    set data_width 64
+  };
+
+  ad_ip_instance spi_engine_execution execution
+  ad_ip_parameter execution CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter execution CONFIG.NUM_OF_CS 1
+  ad_ip_parameter execution CONFIG.NUM_OF_SDI $adc_num_of_channels
+
+  ad_ip_instance axi_spi_engine axi
+  ad_ip_parameter axi CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter axi CONFIG.NUM_OF_SDI $adc_num_of_channels
+  ad_ip_parameter axi CONFIG.NUM_OFFLOAD 1
+  ad_ip_parameter axi CONFIG.ASYNC_SPI_CLK 1
+
+  ad_ip_instance spi_engine_offload offload
+  ad_ip_parameter offload CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter offload CONFIG.NUM_OF_SDI $adc_num_of_channels
+  ad_ip_parameter offload CONFIG.ASYNC_TRIG 1
+  ad_ip_parameter offload CONFIG.ASYNC_SPI_CLK 1
+
+  ad_ip_instance spi_engine_interconnect interconnect
+  ad_ip_parameter interconnect CONFIG.DATA_WIDTH $data_width
+  ad_ip_parameter interconnect CONFIG.NUM_OF_SDI $adc_num_of_channels
+
+  ad_connect axi/spi_engine_offload_ctrl0 offload/spi_engine_offload_ctrl
+  ad_connect offload/spi_engine_ctrl interconnect/s0_ctrl
+  ad_connect axi/spi_engine_ctrl interconnect/s1_ctrl
+  ad_connect interconnect/m_ctrl execution/ctrl
+
+  ad_connect offload/offload_sdi M_AXIS_SAMPLE
+
+  ad_connect execution/spi m_spi
+
+  ad_connect spi_clk offload/spi_clk
+  ad_connect spi_clk offload/ctrl_clk
+  ad_connect spi_clk execution/clk
+  ad_connect clk axi/s_axi_aclk
+  ad_connect spi_clk axi/spi_clk
+  ad_connect spi_clk interconnect/clk
+
+  ad_connect axi/spi_resetn offload/spi_resetn
+  ad_connect axi/spi_resetn execution/resetn
+  ad_connect axi/spi_resetn interconnect/resetn
+
+  ad_connect odr offload/trigger
+
+  ad_connect resetn axi/s_axi_aresetn
+  ad_connect irq axi/irq
+
+current_bd_instance /
+
+# clkgen
+
+ad_ip_instance axi_clkgen axi_cn0561_clkgen
+ad_ip_parameter axi_cn0561_clkgen CONFIG.VCO_DIV 5
+ad_ip_parameter axi_cn0561_clkgen CONFIG.VCO_MUL 48
+ad_ip_parameter axi_cn0561_clkgen CONFIG.CLK0_DIV 10
+
+# dma to receive data stream
+
+ad_ip_instance axi_dmac axi_cn0561_dma
+ad_ip_parameter axi_cn0561_dma CONFIG.DMA_TYPE_SRC 1
+ad_ip_parameter axi_cn0561_dma CONFIG.DMA_TYPE_DEST 0
+ad_ip_parameter axi_cn0561_dma CONFIG.CYCLIC 0
+ad_ip_parameter axi_cn0561_dma CONFIG.SYNC_TRANSFER_START 0
+ad_ip_parameter axi_cn0561_dma CONFIG.AXI_SLICE_SRC 0
+ad_ip_parameter axi_cn0561_dma CONFIG.AXI_SLICE_DEST 1
+ad_ip_parameter axi_cn0561_dma CONFIG.DMA_2D_TRANSFER 0
+ad_ip_parameter axi_cn0561_dma CONFIG.DMA_DATA_WIDTH_SRC [expr $data_width * $adc_num_of_channels]
+ad_ip_parameter axi_cn0561_dma CONFIG.DMA_DATA_WIDTH_DEST 128
```

Comment:
For Zedboard/Cora, the interface should be 64 bits, similar with what the HP ports support.